### PR TITLE
TINY-9219: Fix broken tests on Firefox (#8155)

### PR DESCRIPTION
**Pull request by MitchCTiny**
Related Ticket: TINY-9219

Description of Changes:
* Backport fix to failing tests on Firefox (TINY-9200)

Pre-checks:
* [x] ~Changelog entry added~
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] ~Docs ticket created (if applicable)~

GitHub issues (if applicable):

### CODE REVIEW

**Comment by ephox-mogran** [comment]
This PR noticeably doesn't have the changes to: 

* modules/tinymce/src/core/test/ts/browser/annotate/AnnotationStylingTest.ts
* modules/tinymce/src/models/dom/test/ts/module/table/TableTestUtils.ts

Were they not in 5.10 ?

**Comment by MitchCTiny** [comment]
> This PR noticeably doesn't have the changes to:
> 
> * modules/tinymce/src/core/test/ts/browser/annotate/AnnotationStylingTest.ts
> * modules/tinymce/src/models/dom/test/ts/module/table/TableTestUtils.ts
> 
> Were they not in 5.10 ?

Correct, these files do not exist in 5.10.

**Comment by lnewson** [comment]
> Correct, these files do not exist in 5.10.

Actually you're incorrect here... the latter does exist, it's just part of the table plugin and would likely need fixing: https://github.com/tinymce/tinymce/blob/release/5.10/modules/tinymce/src/plugins/table/test/ts/module/test/TableTestUtils.ts#L55

**Comment by MitchCTiny** [comment]
> > Correct, these files do not exist in 5.10.
> 
> Actually you're incorrect here... the latter does exist, it's just part of the table plugin and would likely need fixing: https://github.com/tinymce/tinymce/blob/release/5.10/modules/tinymce/src/plugins/table/test/ts/module/test/TableTestUtils.ts#L55. The former however you are correct does not exist in 5.x.

Ah thanks for pointing that out, I'll add a commit with those changes.

File: modules/sugar/src/test/ts/browser/LocationTest.ts [hunk 1]
```diff
@@ -136,7 +136,8 @@ UnitTest.asynctest('LocationTest', (success, failure) => {
     pos = SugarLocation.viewport(body);
     assert.eq(0, pos.top);
     assert.eq(0, pos.left);
-    assert.eq(true, scrollBarWidth > 5 && scrollBarWidth < 50 || (platform.os.isOSX() && scrollBarWidth === 0), 'scroll bar width, got=' + scrollBarWidth);
+    const noVisibleScrollbarBrowser = platform.os.isOSX() || (platform.browser.isFirefox() && platform.os.isLinux()) || (platform.browser.isFirefox() && platform.os.isWindows() && platform.os.version.major >= 11);
```
Comment (spocke) [inline modules/sugar/src/test/ts/browser/LocationTest.ts:139]: Maybe we should remove that windows check in this PR. It was added unverified to the previous one and Windows 11 UA detection is not possible. I logged a ticket to look into feature detection instead. So if we ever run the 5.x tests on Win 11 these tests will break.
```suggestion
    const noVisibleScrollbarBrowser = platform.os.isOSX() || (platform.browser.isFirefox() && platform.os.isLinux());
```

File: modules/sugar/src/test/ts/browser/ScrollTest.ts [hunk 2]
```diff
@@ -135,7 +135,8 @@ UnitTest.asynctest('ScrollTest', (success, failure) => {
     const cX = Math.round(center.left);
     const cY = Math.round(center.top);
 
-    assert.eq(true, scrollBarWidth > 5 && scrollBarWidth < 50 || (platform.os.isOSX() && scrollBarWidth === 0), 'scroll bar width, got=' + scrollBarWidth);
+    const noVisibleScrollbarBrowser = platform.os.isOSX() || (platform.browser.isFirefox() && platform.os.isLinux()) || (platform.browser.isFirefox() && platform.os.isWindows() && platform.os.version.major >= 11);
```
Comment (spocke) [inline modules/sugar/src/test/ts/browser/ScrollTest.ts:138]: Same here consider removing the Windows 11 check for now.

**Review by spocke — APPROVED** [review]

**Review by TheSpyder — APPROVED** [review]

**Review by ephox-mogran — APPROVED** [review]

**Pull request merged**